python/triton/compiler/compiler.py

```diff
@@ -164,8 +164,11 @@ def optimize_ttgir(mod, num_stages, num_warps, num_ctas, target, cluster_info, e
     pm.add_tritongpu_remove_layout_conversions_pass()
     pm.add_tritongpu_decompose_conversions_pass()
     pm.add_tritongpu_ws_fixup_missing_attrs_pass()
-    if num_stages != 0:
+    if is_hip() and num_stages != 0:
```

Comment:
The check against `num_stages` makes it not work with the stream pipeliner. Is it intentional?

Reply:
This scheduler is designed to do scheduling for FA kernel with dot slicing enabled. So it's kind of replacing the stream-pipeline pass.